src/chainbin/nnet3-chain-combine.cc

```diff
@@ -102,29 +156,36 @@ int main(int argc, char *argv[]) {
       KALDI_ASSERT(!egs.empty());
     }
 
+    int32 best_n = 1;
```

Comment:
call this best_num_to_combine
and need more documentation in the code-- explain what's going on.  It should be clear to someone who doesn't already understand.